src/Containers/Microsoft.NET.Build.Containers/AuthHandshakeMessageHandler.cs

```diff
@@ -46,25 +51,54 @@ private static bool TryParseAuthenticationInfo(HttpResponseMessage msg, [NotNull
         if (header is { Scheme: "Bearer" or "Basic", Parameter: string bearerArgs })
         {
             scheme = header.Scheme;
-            Dictionary<string, string> keyValues = new();
-            foreach (Match match in BearerParameterSplitter().Matches(bearerArgs))
+            var keyValues = ParseBearerArgs(bearerArgs);
+
+            var result = scheme switch
             {
-                keyValues.Add(match.Groups["key"].Value, match.Groups["value"].Value);
-            }
+                "Bearer" => TryParseBearerAuthInfo(keyValues, out bearerAuthInfo),
+                "Basic" => TryParseBasicAuthInfo(keyValues, msg.RequestMessage!.RequestUri!, out bearerAuthInfo),
+                _ => false
+            };
+            return result;
+        }
+        return false;
 
-            if (keyValues.TryGetValue("realm", out string? realm) && keyValues.TryGetValue("service", out string? service))
+        static bool TryParseBearerAuthInfo(Dictionary<string, string> authValues, [NotNullWhen(true)] out AuthInfo? authInfo) {
+            if (authValues.TryGetValue("realm", out string? realm))
             {
+                string? service = null;
+                authValues.TryGetValue("service", out service);
                 string? scope = null;
-                keyValues.TryGetValue("scope", out scope);
-                authInfo = new AuthInfo(new Uri(realm), service, scope);
+                authValues.TryGetValue("scope", out scope);
+                authInfo = new AuthInfo(realm, service, scope);
                 return true;
             }
+            else {
+                authInfo = null;
+                return false;
+            }
         }
 
-        return false;
+        static bool TryParseBasicAuthInfo(Dictionary<string, string> authValues, Uri requestUri, out AuthInfo? authInfo) {
+            authInfo = null;
```

Comment:
This fixes https://github.com/dotnet/sdk-container-builds/issues/314 and https://github.com/dotnet/sdk-container-builds/issues/466 by looking at the parameters for `Basic` auth.

Reply:
Can you clarify? It looks like no parsing happens here, we just return null bearer info. I guess that's handled a bit below where the scheme is checked before accessing bearerAuthInfo?

Reply:
Sorry, I meant: it fixes those issues by no longer parsing parameters for Basic auth as we don't need them to authenticate.

Reply:
That makes sense, thanks for the clarification!